### PR TITLE
ref(datasets): Add the ability to provide configuration to topic creation

### PR DESCRIPTION
**Pull request by tkaemming**
This allows specifying Kafka configuration parameters for topics at creation time. Similar to #747. This does not alter existing topics.

### CODE REVIEW

**Comment by tkaemming** [comment]
I honestly kind of hate this — this is very brittle when dealing with enforcing expected configuration like topic compaction when topic override flags are used on the command line. At the same time, I don't think anything will work correctly with bootstrap and topic overrides anyway, so it's at least not really a regression.

Tested after running `snuba bootstrap --force` on a fresh volume:

```
(snuba) ted@veneno % docker exec sentry_kafka kafka-topics --zookeeper sentry_zookeeper:2181 --describe --topic events
Topic:events    PartitionCount:1        ReplicationFactor:1     Configs:message.timestamp.type=LogAppendTime
        Topic: events   Partition: 0    Leader: 1001    Replicas: 1001  Isr: 1001
(snuba) ted@veneno % docker exec sentry_kafka kafka-topics --zookeeper sentry_zookeeper:2181 --describe --topic snuba-commit-log
Topic:snuba-commit-log  PartitionCount:1        ReplicationFactor:1     Configs:cleanup.policy=compact
        Topic: snuba-commit-log Partition: 0    Leader: 1001    Replicas: 1001  Isr: 1001
```

We are going to need to alter `message.timestamp.type` on the production `events` topic. The compaction strategy should already be in place on the commit log. We are going to need to figure out how to address configuration updates at some point, as mentioned in #747.

File: snuba/datasets/table_storage.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ class KafkaTopicSpec:
     topic_name: str
     partitions_number: int
     replication_factor: int = 1
+    configuration: Mapping[str, str] = field(default_factory=dict)
```
Comment (fpacifici) [inline snuba/datasets/table_storage.py]: Is it our choice not to provide partitions_number and replication_factor as part of the configuration dictionary but as separate attributes or is it required by the confluent api ?

Comment (tkaemming) [reply snuba/datasets/table_storage.py]: Kafka thing. The partition count and replication factor are set and updated separately from the remainder of the topic configuration in both the admin client and the command line interface.

**Comment by tkaemming** [comment]
I think I am going to need to approach this differently per https://github.com/getsentry/snuba/pull/752#discussion_r375012480.

Going to close this for now, I think the end result will be similar but quite different.

**Pull request closed without merging**